source/rmarkdown/compositional_analysis/test_rademu.Rmd

```diff
@@ -0,0 +1,252 @@
+```
+
+
+
+```{r}
+m_fit <- emuFit(
+  formula = ~ Landgebruik_MBAG + Diepte + Landgebruik_MBAG:Diepte, 
+  Y = physeq_Olig01_Annelida_genus,
+  cluster = sample_data(physeq_Olig01_Annelida_genus)$Cmon_PlotID,
+  run_score_tests = FALSE) 
+
+m_fit$estimation_converged
+```
+
+Visualise:
+
+```{r}
+coeftab <- as_tibble(m_fit$coef)
+
+genustab <- phyloseq::tax_table(physeq_Olig01_Annelida_genus)
+
+genustbl <- as_tibble(
+  genustab@.Data
+) |> 
+  mutate(
+    category = dimnames(genustab)[[1]]) |> 
+  rowwise() |> 
+  mutate(
+    across(
+      all_of(c("phylum", "class", "order", "family", "genus")),
+      \(x) !grepl("_otu", x),
+      .names = "{.col}_"
+    ),
+    index = sum(c_across(phylum_:genus_)),
+    resolved_rank = c("phylum", "class", "order", "family", "genus")[index]
+  ) |> 
+  ungroup() |> 
+  dplyr::select(phylum:category, resolved_rank) |> 
+  mutate(
+    resolved_name = case_when(
+      resolved_rank == "genus" ~ genus,
+      resolved_rank == "family" ~ family,
+      resolved_rank == "order" ~ order,
+      resolved_rank == "class" ~ class,
+      TRUE ~ phylum
+    ),
+    resolved_higher = case_when(
+      resolved_rank == "genus" ~ family,
+      resolved_rank == "family" ~ family,
+      resolved_rank == "order" ~ order,
+      resolved_rank == "class" ~ class,
+      TRUE ~ phylum
+    )
+  )
+
+coeftab |>
+  inner_join(genustbl) |> 
+  mutate(
+    genus = factor(genus, levels = unique(genus[order(resolved_higher)]))
+  ) |> 
+  ggplot() +
+  geom_pointrange(
+    aes(
+      x = genus,
+      y = estimate, ymin = lower, ymax = upper,
+      colour = resolved_higher)
+  ) + 
+  facet_wrap(~covariate)
+
+```
+
+Identify taxa for which you want robust score tests:
+
+For instance, which taxa at depth 0 - 10 are more likely to occur or not occur in Natuurgrasland compared to the reference category Akker:
+
+The following code chunk takes a very long time to run and is therefore not evaluated.
+But it can be run in parallel: https://statdivlab.github.io/radEmu/articles/parallel_radEmu.html
+
+A trial for one taxon took >30 minutes (did not wait to finish) to calculate the score test.
+This would need to be multiplied by the number of taxa of interest to get the time to run the chunk below.
+
+
+
+
+```{r eval=FALSE}
+coefselection <- coeftab |> 
+  filter(
+    covariate == "Landgebruik_MBAGNatuurgrasland",
+    sign(lower) == sign(upper)
+  )
+
+
+covariate_to_test <- which(
+  rownames(m_fit$B) == "Landgebruik_MBAGNatuurgrasland")
+taxa_to_test <- which(
+  coefselection$category %in% colnames(m_fit$B)
+  )
+m_refit <- emuFit(
+  formula = ~ Landgebruik_MBAG + Diepte + Landgebruik_MBAG:Diepte, 
+  Y = physeq_Olig01_Annelida_genus,
+  test_kj = data.frame(
+    k = covariate_to_test, 
+    j = taxa_to_test[1]),
+  fitted_model = m_fit,
+  refit = FALSE,
+  run_score_tests = TRUE
+)
+
+m_refit$coef$pval[taxa_to_test]
+```
+
+Visualise only the significant (p-value <= 0.05) taxa according to the score test:
```

Comment:
Not specific to rademu, but this sounds like a good idea in general? Can it be applied to the sccomp data?

Reply:
Yes, but keep in mind that we have land-use interacting with depth. So the first thing to do is define the specific contrasts we are interested in. Gather for which taxa they are significant and then (optionally) visualise these results (can also be tabularized).